### DISCUSSION TITLE
Hotkey to highlight all NPCs

### DISCUSSION THREAD

**Discussion by z2oh** [runelite/runelite#17025] [Ideas]
When doing quests or exploring new areas I will often add `*` to my "NPCs to Highlight" string to highlight all NPCs. Adding and removing this through the config is a bit tedious, and it would be convenient to toggle this with a key press. The ground items plugin offers very similar functionality: pressing `Alt` (by default) will highlight all items on the ground, including those that are normally hidden by the plugin.

It was actually pretty easy to copy this functionality from the ground items plugin, here's a gif of it in action:
![npc_highlight](https://github.com/runelite/runelite/assets/385684/412aa663-83eb-44af-81f6-375ed792e20a)

Is this something worth making a pull request over? I have reused the `Alt` key by default (but this is configurable). Currently, the global highlight style is used, but I could see it being useful to highlight the extra NPCs in a different color, similarly to how ground items are highlighted in gray. Should this be enabled by default, or should one need to enable this functionality in config?

**Comment by geheur** [comment]
Make it a hub plugin. This is a pretty niche feature.

**Reply by tcpowell** [reply]
https://runelite.net/plugin-hub/show/interactable

**Reply by z2oh** [reply]
This plugin appears to solve this problem, so I'll close the discussion. Thanks!
